src/App.tsx

```diff
@@ -1,61 +1,153 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './components/types/User';
+import { TodoWithUser } from './components/types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedTodos: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(({ id }) => id === todo.userId),
+}));
+
+function findUserById(userId: number): User | undefined {
+  return usersFromServer.find(({ id }) => id === userId);
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserIdError, setIsUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let isError = false;
+
+    if (!title) {
+      setIsTitleError(true);
+      isError = true;
+    }
+
+    if (userId === 0) {
+      setIsUserIdError(true);
+      isError = true;
+    }
+
+    if (isError) {
+      return;
+    }
+
+    function resetForm() {
+      setTitle('');
+      setUserId(0);
+    }
+
+    const todoIds = todos.map(({ id }) => id);
+    const maxTodoId = Math.max(...todoIds);
+
+    const newTodo: TodoWithUser = {
+      id: maxTodoId + 1,
+      title,
+      completed: false,
+      userId,
+      user: findUserById(userId),
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
In order to have an up-to-date state, we must take it from the callback:
```suggestion
    setTodos((currentTodos) => [...currentTodos, newTodo]);
```